pkg/costattribution/manager.go

```diff
@@ -291,17 +310,31 @@ func (m *Manager) updateTracker(userID string) (*SampleTracker, *ActiveSeriesTra
 	newMaxCardinality := m.limits.MaxCostAttributionCardinality(userID)
 	newCooldownDuration := m.limits.CostAttributionCooldown(userID)
 
-	if !st.hasSameLabels(labels) || st.maxCardinality != newMaxCardinality || st.cooldownDuration != newCooldownDuration {
+	if st == nil || !st.hasSameLabels(labels) || st.maxCardinality != newMaxCardinality || st.cooldownDuration != newCooldownDuration {
```

Comment:
If `st` is nil, or `at` is nil, it means that we couldn't create the trackers in the calls above, so maybe it doesn't make sense to go through the entire process and just have an early return above?